### PR TITLE
planner: Fix the issue that may generate many plandigests when the inner table is clustered

### LINKED ISSUE #47634: tpcc prepare different plandigests for the same SQL
## Bug Report
Run tiup tpcc preapre will get different plandigests for the same SQL.
### 1. Minimal reproduce step (Required)
tiup prepare and check the slow query log.

```
SELECT
  count(*)
FROM
  (
    SELECT
      c.c_id,
      c.c_d_id,
      c.c_w_id,
      c.c_balance c1,
      (
        SELECT
          sum(ol_amount)
        FROM
          orders,
          order_line
        WHERE
          OL_W_ID = O_W_ID
          AND OL_D_ID = O_D_ID
          AND OL_O_ID = O_ID
          AND OL_DELIVERY_D IS NOT NULL
          AND O_W_ID = ?
          AND O_D_ID = c.C_D_ID
          AND O_C_ID = c.C_ID
      ) sm,
      (
        SELECT
          sum(h_amount)
        FROM
          history
        WHERE
          H_C_W_ID = ?
          AND H_C_D_ID = c.C_D_ID
          AND H_C_ID = c.C_ID
      ) smh
    FROM
      customer c
    WHERE
      c.c_w_id = ?
  ) t
WHERE
  c1 <> sm - smh
```
### 2. What did you expect to see? (Required)
With the same plandigest
### 3. What did you see instead (Required)
Got the different plandigest
### 4. What is your TiDB version? (Required)

v7.4.0

### PR DESCRIPTION
**Pull request by King-Dylan**
### What problem does this PR solve?
Issue Number: close #47634  
Problem Summary:
After pr:[#38259](https://github.com/pingcap/tidb/pull/38259) queries with the same plan have different plan_digest in information_schema.statement_summary
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by King-Dylan** [comment]
> ### What problem does this PR solve?
> Issue Number: close #47634 Problem Summary: After pr:[#38259](https://github.com/pingcap/tidb/pull/38259) queries with the same plan have different plan_digest in information_schema.statement_summary
> 
> ### What is changed and how it works?
> ### Check List
> Tests
> 
> * [x]  Unit test
> * [ ]  Integration test
> * [ ]  Manual test (add detailed scripts or steps below)
> * [ ]  No need to test
>   > * [ ]  I checked and no code files have been changed.
> 
> Side effects
> 
> * [ ]  Performance regression: Consumes more CPU
> * [ ]  Performance regression: Consumes more Memory
> * [ ]  Breaking backward compatibility
> 
> Documentation
> 
> * [ ]  Affects user behaviors
> * [ ]  Contains syntax changes
> * [ ]  Contains variable changes
> * [ ]  Contains experimental features
> * [ ]  Changes MySQL compatibility
> 
> ### Release note
> Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.
> 
> ```
> Supports generating the same plan digest when accessconds are displayed in the tablescan operator..
> ```

File: pkg/planner/core/explain.go [hunk 1]
```diff
@@ -78,9 +78,17 @@ func (p *PhysicalIndexScan) OperatorInfo(normalized bool) string {
 	var buffer strings.Builder
 	if len(p.rangeInfo) > 0 {
 		if !normalized {
-			buffer.WriteString("range: decided by ")
-			buffer.WriteString(p.rangeInfo)
-			buffer.WriteString(", ")
+			buffer.WriteString("range: decided by [")
+			isFirst := true
+			for _, rangeInfoPart := range p.rangeInfo {
```
Comment (AilinKid) [inline pkg/planner/core/explain.go:83]: you can use `strings.Join(p.rangeInfo, " ")` to concat the entire string.

**Review by qw4990 — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Comment by qw4990** [comment]
PTAL @King-Dylan 
image

**Pull request merged**